packages/fluentui/react-northstar/src/components/Tree/Tree.tsx

```diff
@@ -96,129 +104,130 @@ export interface TreeItemForRenderProps {
-            if (item['expanded'] && acc.indexOf(item['id']) === -1) {
-              acc.push(item['id']);
-            }
-
-            if (item['items']) {
-              return expandedItemsGenerator(item['items'], acc);
-            }
-
-            return acc;
-          },
-          acc,
-        );
 
-      expandedItemsGenerator(items);
-    }
+      if (item['items']) {
+        return expandedItemsGenerator(item['items'], acc);
+      }
 
-    return {
-      activeItemIds,
-      selectedItemIds,
-    };
-  }
+      return acc;
+    },
+    acc,
+  );
+
+const iterateItems = (items, acc = []) =>
+  _.reduce(
+    items,
+    (acc, item) => {
+      if (item['selected'] && acc.indexOf(item['id']) === -1) {
+        acc.push(item['id']);
+      }
+      if (item['items']) {
+        return iterateItems(item['items']);
+      }
 
-  getInitialAutoControlledState() {
-    return { activeItemIds: [], selectedItemIds: [] };
-  }
+      return acc;
+    },
+    acc,
+  );
+
+const Tree: React.FC<WithAsProp<TreeProps>> &
+  FluentComponentStaticProps<TreeProps> & {
+    Item: typeof TreeItem;
+    Title: typeof TreeTitle;
+  } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Tree.displayName, context.telemetry);
+  setStart();
+
+  const {
+    exclusive,
+    items,
+    renderItemTitle,
+    selectable,
+    children,
+    renderedItems,
+    className,
+    design,
+    styles,
+    variables,
+  } = props;
+
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(Tree.handledProps, props);
+
+  const [activeItemIds, setActiveItemIdsState] = useAutoControlled({
+    defaultValue: props.defaultActiveItemIds,
+    value: props.activeItemIds,
+    initialValue: expandedItemsGenerator(items),
+  });
+
+  const [selectedItemIds, setSelectedItemIdsState] = useAutoControlled({
+    defaultValue: props.defaultSelectedItemIds,
+    value: props.selectedItemIds,
+    initialValue: iterateItems(items),
+  });
+
+  const getA11yProps = useAccessibility<TreeBehaviorProps>(props.accessibility, {
+    debugName: Tree.displayName,
+    rtl: context.rtl,
+  });
+
+  const { classes } = useStyles<TreeStylesProps>(Tree.displayName, {
+    className: treeClassName,
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
 
-  treeRef = React.createRef<HTMLElement>();
-  itemsRef = new Map<string, React.RefObject<HTMLElement>>();
+  const treeRef = React.useRef<HTMLElement>();
+  const itemsRef = new Map<string, React.RefObject<HTMLElement>>();
 
-  onFocusParent = (parent: string) => {
-    const parentRef = this.itemsRef.get(parent);
+  const onFocusParent = React.useCallback((parent: string) => {
+    const parentRef = itemsRef.get(parent);
 
     if (!parentRef || !parentRef.current) {
       return;
     }
 
     parentRef.current.focus();
+  }, []);
```

Comment:
```suggestion
  }, [parentRef, itemsRef]);
```

And this is dangerous as `itemsRef` are created on each render. Let's find a way to make `itemsRef` stable.